usaspending_api/data_load/fpds_loader.py

```diff
@@ -0,0 +1,260 @@
+        broker_transactions = _extract_broker_objects(chunk)
+        modified_awards.extend(load_chunk(broker_transactions))
+    return modified_awards
+
+
+def load_chunk(chunk):
+    """
+    Run transaction load for the provided broker data.
+    This will create any new rows in other tables to support the transaction
+    data, but does NOT update "secondary" award values like total obligations or C -> D linkages. If transactions are
+    being reloaded, this will also leave behind rows in supporting tables that won't be removed unless destory_orphans
+    is called.
+    :param chunk: array of DictCursors, representing data from broker to be loaded
+    :return: award id for each award touched
+    """
+    with Timer() as timer:
+        load_objects = _transform_objects(chunk)
+
+        retval = _load_transactions(load_objects)
+    logger.info("ran load in {}".format(str(timer.elapsed)))
+    return retval
+
+
+def _extract_broker_objects(id_list):
+    formatted_id_list = "({})".format(",".join(map(str, id_list)))
+
+    with psycopg2.connect(dsn=BROKER_CONNECTION_STRING) as connection:
+        with connection.cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
+            sql = "SELECT * from detached_award_procurement where detached_award_procurement_id in {}".format(
+                formatted_id_list
+            )
+
+            cursor.execute(sql)
+            results = cursor.fetchall()
+
+    return results
+
+
+def _create_load_object(broker_object, non_boolean_column_map, boolean_column_map, function_map):
+    retval = {}
+    if non_boolean_column_map:
+        retval.update(
+            {non_boolean_column_map[key]: capitalize_if_string(broker_object[key]) for key in non_boolean_column_map}
+        )
+
+    if boolean_column_map:
+        retval.update({boolean_column_map[key]: false_if_null(broker_object[key]) for key in boolean_column_map})
+
+    if function_map:
+        retval.update({key: func(broker_object) for key, func in function_map.items()})
+
+    return retval
+
+
+def _transform_objects(broker_objects):
+    retval = []
+
+    for broker_object in broker_objects:
+        connected_objects = {}
+
+        connected_objects["recipient_location"] = _create_load_object(
+            broker_object, recipient_location_columns, None, recipient_location_functions
+        )
+
+        connected_objects["legal_entity"] = _create_load_object(
+            broker_object, legal_entity_columns, legal_entity_boolean_columns, legal_entity_functions
+        )
+
+        connected_objects["place_of_performance_location"] = _create_load_object(
+            broker_object, place_of_performance_columns, None, place_of_performance_functions
+        )
+
+        # matching award. NOT a real db object, but needs to be stored when making the link in load_transactions
+        connected_objects["generated_unique_award_id"] = broker_object["unique_award_key"]
```

Comment:
Instead of setting this value here, couldn't it be retrieved later from `obj["transaction_fpds"]["unique_award_key"]`

Line 226
```python
load_object["transaction_fpds"]["unique_award_key"]
```